NitroxServer/GameLogic/StoryManager.cs

```diff
@@ -0,0 +1,268 @@
+
+/// <summary>
+/// Keeps track of time and Aurora-related events.
+/// </summary>
+public class StoryManager
+{
+    private readonly Stopwatch stopWatch = new();
+    private readonly PlayerManager playerManager;
+    private readonly PDAStateData pdaStateData;
+    private readonly StoryGoalData storyGoalData;
+    private string seed;
+
+    public double AuroraCountdownTimeMs;
+    public double AuroraWarningTimeMs;
+
+    private double elapsedTimeOutsideStopWatchMs;
+
+    /// <summary>
+    /// Total Elapsed Time in milliseconds
+    /// </summary>
+    public double ElapsedTimeMs
+    {
+        get => stopWatch.ElapsedMilliseconds + elapsedTimeOutsideStopWatchMs;
+        internal set
+        {
+            elapsedTimeOutsideStopWatchMs = value - stopWatch.ElapsedMilliseconds;
+        }
+    }
+
+    /// <summary>
+    /// Total Elapsed Time in seconds
+    /// </summary>
+    public double ElapsedSeconds
+    {
+        get => ElapsedTimeMs * 0.001;
+        private set => ElapsedTimeMs = value * 1000;
+    }
+
+    /// <summary>
+    /// Subnautica's equivalent of days
+    /// </summary>
+    /// <remarks>
+    /// Using ceiling because days count start at 1 and not 0.
+    /// </remarks>
+    public int Day => (int)Math.Ceiling(ElapsedTimeMs / TimeSpan.FromMinutes(20).TotalMilliseconds);
+
+    public Timer ResyncTimer;
+    /// <summary>
+    /// Time in seconds between each resync packet sending.
+    /// </summary>
+    private const int RESYNC_INTERVAL = 60;
+
+    public StoryManager(PlayerManager playerManager, PDAStateData pdaStateData, StoryGoalData storyGoalData, string seed, double elapsedSeconds, double? auroraExplosionTime, double? auroraWarningTime)
+    {
+        this.playerManager = playerManager;
+        this.pdaStateData = pdaStateData;
+        this.storyGoalData = storyGoalData;
+        this.seed = seed;
+        // Default time in Base SN is 480s
+        elapsedTimeOutsideStopWatchMs = elapsedSeconds == 0 ? TimeSpan.FromSeconds(480).TotalMilliseconds : elapsedSeconds * 1000;
+        AuroraCountdownTimeMs = auroraExplosionTime ?? GenerateDeterministicAuroraTime(seed);
+        AuroraWarningTimeMs = auroraWarningTime ?? ElapsedTimeMs;
+        SetupResyncInterval();
+    }
+
+    /// <summary>
+    /// Creates a timer that periodically sends resync packets to players
+    /// </summary>
+    public void SetupResyncInterval()
+    {
+        ResyncTimer = new()
+        {
+            Interval = TimeSpan.FromSeconds(RESYNC_INTERVAL).TotalMilliseconds,
+            AutoReset = true
+        };
+        ResyncTimer.Elapsed += delegate
+        {
+            playerManager.SendPacketToAllPlayers(new AuroraAndTimeUpdate(GetInitialTimeData(), false));
+        };
+    }
+
+    /// <summary>
+    /// Tells the players to start Aurora's explosion event
+    /// </summary>
+    /// <param name="countdown">Wether we should make Aurora explode instantly or after a short countdown</param>
```

Comment:
```suggestion
    /// <param name="countdown">Whether we should make Aurora explode instantly or after a short countdown</param>
```

Noticed a little oopsie in your spelling :)